ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh

```diff
@@ -13,31 +13,82 @@ CONFIG="${SHARED_DIR}/install-config.yaml"
 
 expiration_date=$(date -d '8 hours' --iso=minutes --utc)
 
+declare -g MASTER_FAMILY
+declare -g MASTER_TYPE
+
 function join_by { local IFS="$1"; shift; echo "$*"; }
 
+function get_master_size {
+  master_size=null
+  if [[ "${SIZE_VARIANT}" == "xlarge" ]]; then
+    master_size=8xlarge
+  elif [[ "${SIZE_VARIANT}" == "large" ]]; then
+    master_size=4xlarge
+  elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
+    master_size=2xlarge
+  fi
+  echo ${master_size}
+}
+
+function set_master_type {
+  local size
+  master_type=null
+  size=$(get_master_size)
+  if [[ "${size}" == "null" ]]; then
+    MASTER_TYPE="${size}"
+    return
+  fi
+  MASTER_TYPE="${MASTER_FAMILY}.$(get_master_size)"
+}
+
+function choose_preferred_type {
+  local type_preferred
+  local type_backup
+  local zones_cnt_pref
+  local zones_cnt_backup
+
+  type_preferred="${1}.$(get_master_size)"
+  type_backup="${2}.$(get_master_size)"
+
+  zones_cnt_pref=$(aws ec2 describe-instance-type-offerings --region "${REGION}" --location-type availability-zone --filters Name=instance-type,Values="${type_preferred}" --query 'InstanceTypeOfferings[].Location' --output json | jq -r '. | length')
+  zones_cnt_backup=$(aws ec2 describe-instance-type-offerings --region "${REGION}" --location-type availability-zone --filters Name=instance-type,Values="${type_backup}" --query 'InstanceTypeOfferings[].Location' --output json | jq -r '. | length')
```

Comment:
This does not take into account whether AZs are actually available or whether the AZs were specified already in the install-config.yaml.

Reply:
you're right. I just fixed it:

> AWS just change my use case from `sa-east-1`, changing the region to `af-south-1` will get the expected results

``` bash
$ yq .controlPlane.platform.aws install-config.yaml
{
  "zones": [
    "af-south-1a",
    "af-south-1b"
  ]
}

$ export LEASED_RESOURCE=af-south-1
$ bash  ipi-conf-aws-commands.sh
zones already set in install-config.yaml, skipped

$ yq .controlPlane.platform.aws install-config.yaml
{
  "zones": [
    "af-south-1a",
    "af-south-1b"
  ],
  "type": "m5.xlarge"
}

$ export LEASED_RESOURCE=us-east-1
$ cp install-config.yaml-use1 install-config.yaml
$ yq .controlPlane.platform.aws install-config.yaml
{
  "zones": [
    "us-east-1a",
    "us-east-1b"
  ]
}

$ bash  ipi-conf-aws-commands.sh
zones already set in install-config.yaml, skipped

$ yq .controlPlane.platform.aws install-config.yaml
{
  "zones": [
    "us-east-1a",
    "us-east-1b"
  ],
  "type": "m6i.xlarge"
}

```